src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.service.ts

```diff
@@ -6,47 +6,45 @@ import { DotLayoutColumn } from '../../shared/models/dot-layout-column.model';
 import * as _ from 'lodash';
 import { DotPageView } from '../../shared/models/dot-page-view.model';
 import { NgGridItemConfig } from 'angular2-grid';
+import { DOT_LAYOUT_GRID_NEW_ROW_TEMPLATE } from '../../shared/models/dot-layout.const';
 
 /**
- * Provide methods to transform data types, to be able to display it in the NgGrid.
+ * Provide methods to transform NgGrid model into PageView model and viceversa.
  *
- * @class DotEditLayoutGridService
+ * @class DotEditLayoutService
  */
 @Injectable()
-export class DotEditLayoutGridService {
+export class DotEditLayoutService {
     constructor() {}
 
     /**
-     * Transform and Object from  DotPageView to DotLayoutGridBox to be able to display it on the Grid
+     * Transform and return DotPageView to DotLayoutGridBox
      *
      * @param {DotPageView} dotPageView
      * @param {NgGridItemConfig} newRow
      * @returns {DotLayoutGridBox[]}
      */
-    getDotLayoutGridBox(dotPageView: DotPageView, newRow: NgGridItemConfig): DotLayoutGridBox[] {
+    getDotLayoutGridBox(dotPageView: DotPageView): DotLayoutGridBox[] {
```

Comment:
Make sure you pass to methods exactly what they need, more params means most complex method and harder to test.